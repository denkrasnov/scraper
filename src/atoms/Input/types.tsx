import { ChangeEvent } from "react";

export interface InputProps {
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  placeholder: string;
  value?: string;
  name: string;
}
