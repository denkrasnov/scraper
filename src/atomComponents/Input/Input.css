.Input {
  padding: 12px;
  height: 48px;
  width: 100%;
  border: 0;
  border-radius: 4px; 
  box-shadow: 0 1px 5px rgba(8, 31, 44, 0.12);
  outline: none;
}