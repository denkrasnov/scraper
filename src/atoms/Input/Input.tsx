import React, { FC } from "react";

import { InputProps } from "./types";
import styles from "./Input.css";

const Input: FC<InputProps> = props => {
  const { onChange, placeholder, value } = props;
  return (
    <input
      className={styles.Input}
      onChange={onChange}
      placeholder={placeholder}
      type="text"
      value={value}
    />
  );
};

export default Input;
