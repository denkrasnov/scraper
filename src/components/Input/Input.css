.Input {
  padding: 12px;
  height: 48px;
  width: 100%;
  border: 0;
  border-radius: 4px; 
  box-shadow: 0 1px 5px var(--shadow-color);
  outline: none;
}