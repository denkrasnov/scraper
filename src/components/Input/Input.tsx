import React, { FC } from "react";

import { InputProps } from "./types";
import styles from "./Input.css";

const Input: FC<InputProps> = props => {
  const { onChange, placeholder, value } = props;
  return (
    <input
      className={styles.Input}
      onChange={onChange}
      type="text"
      value={value}
      placeholder={placeholder}
    />
  );
};

export default Input;
