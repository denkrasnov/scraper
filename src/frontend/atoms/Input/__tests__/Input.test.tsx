import { shallow } from "enzyme";

import Input from "..";

describe("Input", () => {
  it("should render", () => {
    const component = shallow(
      <Input name="__NAME__" placeholder="__PLACEHOLDER__" />
    );
    expect(component).toMatchSnapshot();
  });
});
