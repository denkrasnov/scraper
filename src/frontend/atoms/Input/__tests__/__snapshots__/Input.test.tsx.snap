// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render 1`] = `
<input
  className="Input"
  name="__NAME__"
  placeholder="__PLACEHOLDER__"
  type="text"
/>
`;
